src/charm.py

```diff
@@ -75,63 +73,55 @@ def _handle_event(self, event: ops.EventBase) -> None:  # noqa: C901
             self._ensure_installed()
             config = self._get_config()
             self._mount_filesystems(config)
-        except StopCharmError:
+        except StopCharmError as e:
             # This was the cleanest way to ensure the inner methods can still return prematurely
             # when an error occurs.
+            self.app.status = e.status
             return
 
-        self.unit.status = ops.ActiveStatus("Mounted shares.")
+        self.unit.status = ops.ActiveStatus("Mounted filesystems.")
 
     def _ensure_installed(self):
         """Ensure the required packages are installed into the unit."""
         if not self._mounts_manager.installed:
             self.unit.status = ops.MaintenanceStatus("Installing required packages.")
-            self._mounts_manager.ensure(apt.PackageState.Present)
+            self._mounts_manager.install()
 
     def _get_config(self) -> dict[str, dict[str, str | bool]]:
         """Get and validate the configuration of the charm."""
         try:
-            config = json.loads(str(self.config.get("mountinfo", "")))
+            config = json.loads(str(self.config.get("mounts", "")))
             validate(config, CONFIG_SCHEMA)
             config: dict[str, dict[str, str | bool]] = config
             for fs, opts in config.items():
                 for opt in ["noexec", "nosuid", "nodev", "read-only"]:
                     opts[opt] = opts.get(opt, False)
             return config
         except (json.JSONDecodeError, ValidationError) as e:
-            self.app.status = ops.BlockedStatus(
-                f"invalid configuration for option `mountinfo`. reason: {e}"
+            raise StopCharmError(
```

Comment:
Add a log.error message above this please (so the user can refer to the error message)